solver/cvrp.py

```diff
@@ -409,31 +409,36 @@ def check_feasibility(solution):
 
             return True
 
-        num_states = model.states.size()
-        solution = []
-        for i in range(num_states):
-            # extract the solution
-            decision = next(model.iter_decisions())
-            solution_candidate = [[int(v) + 1 for v in route.state(i)] for route in list(decision.iter_successors())]
-
-            solver_objective = model.objective.state(i)
-            assert abs(solver_objective - recompute_objective(solution_candidate)) < tolerance
-
-            solver_feasibility = True
-            for c in model.iter_constraints():
-                if c.state(i) < 0.5:
-                    solver_feasibility = False
-
-            # Check feasibility and if feasible, break
-            assert solver_feasibility == check_feasibility(solution_candidate)
-            if not solver_feasibility:
-                print(f"Sample {i} is infeasible")
-            else:
-                solution = solution_candidate
-                break
+
+        def get_solution():
```

Comment:
This function is getting a bit too big, I think. `get_solution()` could be a method to the CVRP class, as well as `check_feasibility` and `recompute_objective`, perhaps adding a leading underscore to them to show that they're only used internally.

Reply:
Let me know if this works: https://github.com/dwave-examples/mvrp/pull/12/commits/0b9a304e6e4c8ed2bbd7e902200748f4dde018a8

Reply:
Thanks @k8culver